sunpy/coordinates/tests/test_transformations.py

```diff
@@ -0,0 +1,45 @@
+from sunpy.coordinates import Helioprojective
+
+import astropy.units as u
+from astropy.tests.helper import quantity_allclose
+
+import numpy as np
+
+
+def test_hpc_hpc():
+    rsun = 1*u.m
+    D0 = 1*u.km
+    L0 = 1*u.deg
+
+    hpc_in = Helioprojective(0*u.arcsec, 0*u.arcsec, rsun=rsun, D0=D0)
+    hpc_out = Helioprojective(L0=1*u.deg, D0=D0, rsun=rsun)
+
+    hpc_new = hpc_in.transform_to(hpc_out)
+
+    assert hpc_new.L0 == hpc_out.L0
+    assert hpc_new.B0 == hpc_out.B0
+    assert hpc_new.D0 == hpc_out.D0
+
+    # Calculate the distance subtended by an angle of L0 from the centre of the
+    # Sun.
+    dd = -1 * rsun * np.tan(L0)
```

Comment:
Wait... I think I'm starting to get the picture...

If L0 is the angel between the two "observers", and you say that `dd` is the distance between the observers. Right?
If it's so... what do you need `rsun` for? Ooohh.. I see, it's the distance in the sun surface. So, the minus is because the original one is `dd` meters behind (to the East) than the new one. Right?

So, just a stupid question... in the `theta` calculation, should not be more precise to use `arctan(dd/(D0 - rsun))`?
And by the way, would not be better to use `arctan2`?